### PR TITLE
feat(financial-statement-cemetery): Make financial statement cemetery application

### PR DESCRIPTION
**Pull request by jonnigs**
## What

Make financial-statemant-cemetery into a separate application instead of having it a part of financial-statements-inao 

## Why

Financial-statements-inao are three similar but quite different applications in one. Two of them are causing errors, so this PR is trying to fix that. Most of the code is copy/paste from the old application, logic was simplified in cases where we don't have to worry about the other two cases any more.

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/application/templates/financial-statement-cemetery/src/lib/dataSchema.ts [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+    }),
+  fullName: z.string().refine((x) => !!x, { params: m.required }),
+  powerOfAttorneyNationalId: z.string().optional(),
+  powerOfAttorneyName: z.string().optional(),
+  phoneNumber: z.string().refine(
+    (p) => {
+      // ignore validation on dev to test with Gervimenn remove b4 release
+      if (isRunningOnEnvironment('dev')) {
+        return true
+      }
```
Comment (Toti91) [inline libs/application/templates/financial-statement-cemetery/src/lib/dataSchema.ts:121]: Remove?

File: libs/application/templates/financial-statement-cemetery/src/lib/financialStatementCemeteryTemplate.ts [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+  ApplicationStateSchema<Events>,
+  Events
+> = {
+  type: ApplicationTypes.FINANCIAL_STATEMENT_CEMETERY,
+  name: m.applicationTitle,
+  institution: m.institutionName,
+  translationNamespaces: [configuration.translation],
+  dataSchema,
+  allowedDelegations: [{ type: AuthDelegationType.ProcurationHolder }],
+  stateMachineConfig: {
```
Comment (Toti91) [inline libs/application/templates/financial-statement-cemetery/src/lib/financialStatementCemeteryTemplate.ts:43]: We should probably add a feature flag for this

**Review by Toti91 — CHANGES_REQUESTED** [review]
Couple of things

File: libs/application/templates/financial-statement-cemetery/src/lib/dataSchema.ts (hunk 1 above)
Comment (jonnigs) [reply libs/application/templates/financial-statement-cemetery/src/lib/dataSchema.ts:121]: Probably, this is in the inao version as well...

File: libs/application/templates/financial-statement-cemetery/src/forms/prerequisites/prerequsitesSection.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        buildDataProviderItem({
+          provider: CurrentUserTypeProvider,
+          title: m.dataCollectionUserFinancialInfoTitle,
+          subTitle: m.dataCollectionUserFinancialInfo,
+        }),
+      ],
+      submitField: buildSubmitField({
+        id: 'submit',
+        placement: 'footer',
+        title: 'asdf',
```
[the remark this answers is not in the document]
Comment (jonnigs) [reply libs/application/templates/financial-statement-cemetery/src/forms/prerequisites/prerequsitesSection.ts:44]: No, but I'll remove

**Review by Toti91 — APPROVED** [review]

File: libs/application/template-api-modules/src/lib/modules/templates/financial-statement-cemetery/financial-statement-cemetery.service.ts [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+      const file = await this.s3
+        .getObject({
+          Bucket: uploadBucket,
+          Key: key,
+        })
+        .promise()
+      const fileContent = file.Body as Buffer
+      return fileContent.toString('base64') || ''
+    } catch (error) {
+      throw new Error('Villa kom upp við að senda umsókn')
```
Comment (FridrikMProgramm) [inline libs/application/template-api-modules/src/lib/modules/templates/financial-statement-cemetery/financial-statement-cemetery.service.ts:85]: Might be an unhelpful error text, maybe, 'Villa kom upp vid að sækja viðhengi'? or something similar in English.

File: libs/application/template-api-modules/src/lib/modules/templates/financial-statement-cemetery/mappers/mapValuesToUserType.ts [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+    },
+  ]
+
+  if (contactsAnswer) {
+    contactsAnswer.map((x) => {
+      const contact: Contact = {
+        nationalId: x.nationalId,
+        name: x.name,
+        contactType:
+          x.role === 'Stjórnarmaður'
```
Comment (FridrikMProgramm) [inline libs/application/template-api-modules/src/lib/modules/templates/financial-statement-cemetery/mappers/mapValuesToUserType.ts:112]: Addd 'Stjórnarmaður' as a constant same with 'Einstaklingur' here earlier

File: libs/application/templates/financial-statement-cemetery/src/fields/CemeteryOperation/CemeteryIncome.tsx [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+      values,
+      CEMETERYOPERATIONIDS.grantFromTheCemeteryFund,
+    )
+
+    if (data?.financialStatementsInaoTaxInfo) {
+      if (!careIncome) {
+        setValue(
+          CEMETERYOPERATIONIDS.careIncome,
+          data.financialStatementsInaoTaxInfo
+            ?.find((x) => x.key === 300)
```
Comment (FridrikMProgramm) [inline libs/application/templates/financial-statement-cemetery/src/fields/CemeteryOperation/CemeteryIncome.tsx:47]: Put the key in a constant?
Same with 301 and 302

File: package.json [hunk 7]
```diff
@@ -363,6 +363,9 @@
     "@storybook/react-webpack5": "7.6.9",
     "@storybook/source-loader": "7.0.20",
     "@storybook/theming": "7.0.20",
+    "@swc-node/register": "~1.9.1",
+    "@swc/core": "~1.5.7",
+    "@swc/helpers": "~0.5.11",
```
Comment (baering) [inline package.json:368]: This seems to be included when generating new projects after the nx upgrade. Could you pin the versions please?

**Review by baering — CHANGES_REQUESTED** [review]
Core files LGTM ✅ 

One minor request.

File: libs/application/templates/financial-statement-cemetery/src/forms/done/index.ts [hunk 8]
```diff
@@ -0,0 +1,10 @@
+import { buildForm } from '@island.is/application/core'
+import { Form, FormModes } from '@island.is/application/types'
+import { conclusionSection } from './conclusionSection'
+
+export const done: Form = buildForm({
+  id: 'done',
+  title: 'Umsókn móttekin',
```
Comment (FridrikMProgramm) [inline libs/application/templates/financial-statement-cemetery/src/forms/done/index.ts:7]: Constant?

File: libs/application/templates/financial-statement-cemetery/src/lib/financialStatementCemeteryTemplate.ts [hunk 9]
```diff
@@ -0,0 +1,149 @@
...
+          lifecycle: pruneAfterDays(60),
+          roles: [
+            {
+              id: Roles.APPLICANT,
+              formLoader: () =>
+                import('../forms/applicationForm').then((val) =>
+                  Promise.resolve(val.FinancialStatementCemeteryForm),
+                ),
+              actions: [
+                { event: 'SUBMIT', name: 'Staðfesta', type: 'primary' },
```
Comment (FridrikMProgramm) [inline libs/application/templates/financial-statement-cemetery/src/lib/financialStatementCemeteryTemplate.ts:105]: Constant for 'Staðfesta'?

**Review by FridrikMProgramm — APPROVED** [review]
LGTM - Some very minor suggestions.

**Pull request merged**